RecoEcal/EgammaClusterAlgos/src/CosmicClusterAlgo.cc

```diff
@@ -122,7 +122,8 @@ std::vector<reco::BasicCluster> CosmicClusterAlgo::makeClusters(
     
   }
   
-   sort(seeds.begin(), seeds.end(), EcalRecHitLess());
+   // sort by descending energy
+   sort(seeds.begin(), seeds.end(), greaterEnergy);
```

Comment:
We were discussing that before, and there seem to be different opinions. Before it was with the lambda, and I got this comment by @slava77:

> In this and apparently many similar cases, I find this a bad coding pattern: a single named method is replaced by non-trivial inlined code [lambdas]. This is going to make the code much less maintainable.

I tend to agree with @ianna actually, because I think code is more maintainable if it's directly clear in the sort function what it does without checking functions in different files. Take for example the previous implementation with the `EcalRecHitLess` functor, which was by definition more a "greater than" function. Reading any code where it's used makes it absolutely not clear that RecHits will be sorted by ascending energy. This can't happen with the unnamed lambda functions. It takes some abstraction away, therefore I think the code gets more maintainable actually.

However, I don't know how that affects compilation time and performance.

[1] https://github.com/cms-sw/cmssw/blob/master/RecoEcal/EgammaCoreTools/interface/EcalRecHitLess.h